include/seqan3/search/algorithm/configuration/utility.hpp

```diff
@@ -0,0 +1,437 @@
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \brief Provides functionality to access get function by enum values.
+ * \author Christopher Pockrandt <christopher.pockrandt AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <meta/meta.hpp>
+
+#include <seqan3/core/algorithm/configuration.hpp>
+#include <seqan3/core/detail/reflection.hpp>
+
+namespace seqan3::search_cfg
+{
+
+/*!\brief Specifies an id for every configuration element.
+ * \ingroup configuration
+ *
+ * \details
+ *
+ * The seqan3::search_cfg::id is used to identify a specific search configuration element independent of
+ * it's concrete type and position within the seqan3::search_cfg::search_configuration object.
+ * Thus one can access the value of the corresponding configuration element via the special get interface.
+ *
+ * ### Example
+ *
+ * ```cpp
+ * search_cfg::search_configuration cfg = search_cfg::max_total_errors(3);
+ * auto max_total_errors = get<search_cfg::id::max_total_errors>(cfg);  // max_total_errors = 3;
+ * ```
+ */
+enum struct id : uint8_t
+{
+    //!\brief Identifier for max_errors configuration.
+    max_error,
+    max_error_rate,
+    on_hit,
+    return_index_iterator,
+    return_text_position,
+    strategy_all,
+    strategy_all_best,
+    strategy_best,
+    strategy_strata,
+    //!\cond
+    // ATTENTION: Must always be the last item; will be used to determine the number of ids.
+    SIZE
+    //!\endcond
+};
+
+} // namespace seqan3::search_cfg
+
+namespace seqan3::detail
+{
+// ----------------------------------------------------------------------------
+// SEQAN3_INVALID_CONFIG
+// ----------------------------------------------------------------------------
+
+//!\cond
+// A common error message for testing valid configuration combinations.
+#define SEQAN3_INVALID_CONFIG(e) "Configuration error: The configuration <" #e "> is not combinable with "  \
```

Comment:
This macro already exists in alignment/configuration/utility. 
Best to refactor it into core/algorithm

Reply:
That's a TODO for later